fdbserver/MasterInterface.h

```diff
@@ -184,26 +189,50 @@ struct GetCommitVersionRequest {
 	}
 };
 
+struct GetTlogPrevCommitVersionReply {
+	constexpr static FileIdentifier file_identifier = 16683183;
+	VersionVector ssVersionVector;
+	GetTlogPrevCommitVersionReply() {}
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, ssVersionVector);
+	}
+};
+
+struct GetTlogPrevCommitVersionRequest {
+	constexpr static FileIdentifier file_identifier = 16683184;
+	std::set<Tag> tagSet;
+	ReplyPromise<GetTlogPrevCommitVersionReply> reply;
+	GetTlogPrevCommitVersionRequest() {}
+	GetTlogPrevCommitVersionRequest(std::set<Tag> tagSet)
+	  : tagSet(tagSet) {}
+	template <class Ar>
+	void serialize(Ar& ar) {
+		serializer(ar, tagSet, reply);
+	}
+};
+
 struct ReportRawCommittedVersionRequest {
 	constexpr static FileIdentifier file_identifier = 1853148;
 	Version version;
 	bool locked;
 	Optional<Value> metadataVersion;
 	Version minKnownCommittedVersion;
-
+	Optional<VersionVector> ssVersionVector;
```

Comment:
As commented above, make this "std::set<Tag>"?

And, why is this made optional? Don't mutations always have a tag(s) associated with them? Are there any exceptions to this?

Reply:
Empty commit batch has no mutations. Not sure if there are other reasons for using `Optional`.

Reply:
Empty commit batch - is that a special kind of transaction? User write transactions can't have no mutations?

Reply:
If the knob is not set, there won't be a version vector. Sending a "not present" object seems cleaner to me when the knob is not set.

I liked the version vector as a container for tags, but agree it has extra information (we don't need a version per tag). I'll replace it with `Set<Tag>`.

Reply:
> Empty commit batch - is that a special kind of transaction? User write transactions can't have no mutations?

The commit proxy will generate an empty batch if there is no transaction received about `MAX_COMMIT_BATCH_INTERVAL` period. It could also be that all transactions conflicted (unlikely though), then an empty transaction batch is created.